tests/testthat/_snaps/tree.md

```diff
@@ -0,0 +1,256 @@
+            ├─id:"c2"
+            └─val:1
+
+# Atomic arrays have sensible defaults with truncation added for longer than 10-elements
+
+    Code
+      tree(list(name = "vectored list", num_vec = 1:10, char_vec = letters))
+    Output
+      {list}
+      ├─name:"vectored list"
+      ├─num_vec:1,2,3,4,5,6,7,8,9,10
+      └─char_vec:"a","b","c","d","e","f","g","h","i","j",...(n = 26)
+
+# Works with HTML tag structures
+
+    Code
+      tree(shiny::sliderInput("test", "Input Label", 0, 1, 0.5))
+    Output
+      {shiny.tag}
+      ├─name:"div"
+      ├─attribs:{list}
+      │ └─class:"form-group shiny-input-container"
+      └─children:[list]
+        ├─1:{shiny.tag}
+        │ ├─name:"label"
+        │ ├─attribs:{list}
+        │ │ ├─class:"control-label"
+        │ │ ├─id:"test-label"
+        │ │ └─for:"test"
+        │ └─children:[list]
+        │   └─1:"Input Label"
+        └─2:{shiny.tag}
+          ├─name:"input"
+          ├─attribs:{list}
+          │ ├─class:"js-range-slider"
+          │ ├─id:"test"
+          │ ├─data-skin:"shiny"
+          │ ├─data-min:"0"
+          │ ├─data-max:"1"
+          │ ├─data-from:"0.5"
+          │ ├─data-step:"0.01"
+          │ ├─data-grid:"true"
+          │ ├─data-grid-num:10
+          │ ├─data-grid-snap:"false"
+          │ ├─data-prettify-separator:","
+          │ ├─data-prettify-enabled:"true"
+          │ ├─data-keyboard:"true"
+          │ └─data-data-type:"number"
+          └─children:[list]
+
+# Max depth and length can be enforced
+
+    Code
+      deep_list <- list(list(id = "b", val = 1, children = list(list(id = "b1", val = 2.5),
+      list(id = "b2", val = 8, children = list(list(id = "b21", val = 4))))), list(
+        id = "a", val = 2))
+      tree(deep_list, max_depth = 1)
+    Output
+      [list]
+      ├─1:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      └─2:{list}
+        ├─id:"a"
+        └─val:2
+    Code
+      tree(deep_list, max_depth = 2)
+    Output
+      [list]
+      ├─1:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      │   ├─1:{list}
+      │   └─2:{list}
+      └─2:{list}
+        ├─id:"a"
+        └─val:2
+    Code
+      tree(deep_list, max_depth = 3)
+    Output
+      [list]
+      ├─1:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      │   ├─1:{list}
+      │   │ ├─id:"b1"
+      │   │ └─val:2.5
+      │   └─2:{list}
+      │     ├─id:"b2"
+      │     ├─val:8
+      │     └─children:[list]
+      └─2:{list}
+        ├─id:"a"
+        └─val:2
+    Code
+      tree(deep_list, max_length = 0)
+      tree(deep_list, max_length = 2)
+    Output
+      [list]
+      ├─1:{list}
+    Code
+      tree(deep_list, max_depth = 1, max_length = 4)
+    Output
+      [list]
+      ├─1:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+
+# Attributes are properly displayed as special children nodes
+
+    Code
+      list_w_attrs <- structure(list(structure(list(id = "a", val = 2), level = 2,
```

Comment:
I'd suggest simplify this example to hone in on exactly the property you are testing.

Reply:
This is a test that catches a bug I had previously in my branch-drawing logic. After a refactor the cause of the bug disappeared but I left the test. Not sure if the test should be removed or not.

Reply:
I'd suggest removing it.